File: entry_level/src/entry_level/entry_level/subscriber1.py
import rclpy
from rclpy.node import Node

from std_msgs.msg import String

class Subscriber(Node):
    def __init__(self):
        super().__init__('subscriber1')

        self.subscriber_ = self.create_subscription(String, 'sensor1_velocity', self.listener_callback, 10)

    def listener_callback(self, msg):
        self.get_logger().info(f"Received: {msg.data}")

def main(arg=None):

    rclpy.init(args=None)
    subscriber1 = Subscriber()

    try:
        rclpy.spin(subscriber1)
    except KeyboardInterrupt:
        subscriber1.get_logger().info("Keyboard Interrupt (SIGINT)")
    finally:
        subscriber1.destroy_node()
        if rclpy.ok():
            rclpy.shutdown()

File: entry_level/src/publisher_cpp/src/subscriber1.cpp
#include "iostream"

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class Subscriber : public rclcpp::Node
{
public: 
    Subscriber()
    : Node("velocity_sub")
    {
        auto topic_callback =
        [this](std_msgs::msg::String::UniquePtr msg) -> void {
            RCLCPP_INFO(this->get_logger(), "Received: '%s'", msg->data.c_str());
        };
        subscription_ =
            this->create_subscription<std_msgs::msg::String>("velocity", 10, topic_callback);
    }

private:
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;
};

int main(int argc, char * argv[])
{
    try
    {
        rclcpp::init(argc, argv);
        rclcpp::spin(std::make_shared<Subscriber>());
        rclcpp::shutdown();
    }
    catch (const std::exception & e)
    {
        std::cerr << "Exception: " << e.what() << std::endl;
    
        rclcpp::shutdown();
        
        return 1;
    }
}